lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'license_detector.dart';
+
+@sealed
+class License {
```

Comment:
Maybe this abstraction is slightly off - it is often being used to represent the tokenized `content` rather than the actual license.

We should probable split it into two classes:

```
class TokenizedContent {
  final String content;
  final List<Token> tokens;
  final List<String, int> frequencyTable;
  factory TokenizedContent(String content) {
    final tokens = tokenize(content);
    final frequencyTable = generateFrequencyTable(tokens);
    return TokenizedContent._(content, tokens, frequencyTable);
  }
 }

class License {
  final String identifier;
  Licence(this.identifier);
}
```

Reply:
If we split it like that then, the `tokenizedContent` will not be associated with its identifier. Should we wrap both of them in other class or make something like `pair`.

Reply:
Yeah I think it is better with a pair. For the input string you don't know the license. So it seems wrong to associate it with an empty string.

You could also do:
```
class License { 
  final String identifier;
  final TokenizedContent tokenizedContent;
  Licence(this.identifier, tokenizedContent);

}
```